test/arrays.coffee

```diff
@@ -26,6 +26,40 @@ test "incorrect indentation without commas", ->
   ok result[0][0] is 'a'
   ok result[1]['b'] is 'c'
 
+# Elisions
+test "array elisions", ->
+  eq [,1].length, 2
+  eq [,,1,2,,].length, 5
+
+  arr = [1,2,3,4,5,6,7,8,9]
+  [,a] = arr
+  [,,,b] = arr
+  arrayEq [a,b], [2,4]
+  [,a,,b,,c,,,d] = arr
+  arrayEq [a,b,c,d], [2,4,6,9]
+  [
+    ,a,
+    ,b,
+    ,c,
+    ,,d] = arr
+  arrayEq [a,b,c,d], [2,4,6,9]
+
+  foo = ([,a]) -> a
+  a = foo arr
+  eq a, 2
+  foo = ([,,,a]) -> a
+  a = foo arr
+  eq a, 4
+  foo = ([,a,,b,,c,,,d]) -> [a,b,c,d]
+  [a,b,c,d] = foo arr
+  arrayEq [a,b,c,d], [2,4,6,9]
+
+  [,a,,,b...] = arr
+  arrayEq [a,b], [2,[5,6,7,8,9]]
+  [,a,...,,b,,c] = arr
+  arrayEq [a,b,c], [2,7,9]
+  [...,a,,,b,,,] = arr
+  arrayEq [a,b], [4,7]
```

Comment:
Shouldn't this be `[3,6]`?

Reply:
```javascript
// arr = [1,2,3,4,5,6,7,8,9]
// arr.length = 9
// [...,a,,,b,,,] = arr

var a, b, i;
i = arr.length - 6, a = arr[i++], i++, i++, b = arr[i++], i++, i++;
// i++ is used to skip 'Elisions'
```
`arr.length - 6` because `[a,,,b,,,].length is 6`
```javascript
i = arr.length - 6 = 9 - 6 = 3
a = arr[i++] = arr[3] = 4
// i = 4
i++, i++ 
// i = 6
b = arr[i++] = arr[6] = 7
// i = 7
i++, i++
// i = 9

```